app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,19 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_posts_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
```

Comment:
- [ ] [OPTIONAL] You did well with this implementation but you could also consider including a `params` method within your `post_controller`. Here is a [guide](`https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides`)